consensus/src/consensus/message.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::any::Any;
+
+use snarkos_storage::{SerialBlock, SerialRecord, SerialTransaction};
+use tokio::sync::oneshot;
+
+use crate::{error::ConsensusError, Address, PrivateKey};
+
+pub struct CreateTransactionRequest {
+    pub old_records: Vec<SerialRecord>,
+    pub old_account_private_keys: Vec<PrivateKey>,
+    pub new_record_owners: Vec<Address>,
+    pub new_birth_program_ids: Vec<Vec<u8>>,
+    pub new_death_program_ids: Vec<Vec<u8>>,
+    pub new_is_dummy_flags: Vec<bool>,
+    pub new_values: Vec<u64>,
+    pub new_payloads: Vec<Vec<u8>>,
+    pub memo: [u8; 32],
+}
+pub struct CreatePartialTransactionRequest {
+    // TransactionKernel
+    pub kernel: Box<dyn Any + Send + Sync>,
+}
+
+pub struct TransactionResponse {
+    pub records: Vec<SerialRecord>,
+    pub transaction: SerialTransaction,
+}
+
+pub enum ConsensusMessage {
+    ReceiveTransaction(Box<SerialTransaction>, oneshot::Sender<bool>),
```

Comment:
The purpose of boxing here is to keep the enum variant sizes aligned, right?

Reply:
aligned isn't the right term, just keep the size small

Reply:
Yeah, meant aligned in a colloquial way, that the variants don't differ too much in size :+1:.